implementations/rust/ockam/ockam_command/src/project/show.rs

```diff
@@ -40,32 +42,113 @@ impl ShowCommand {
 }
 
 async fn rpc(ctx: Context, (opts, cmd): (CommandGlobalOpts, ShowCommand)) -> miette::Result<()> {
-    run_impl(&ctx, opts, cmd).await
+    run_impl(ctx, opts, cmd).await
+}
+
+async fn run_impl(ctx: Context, opts: CommandGlobalOpts, cmd: ShowCommand) -> miette::Result<()> {
+    ProjectShowTui::run(ctx, opts, cmd.name).await
 }
 
-async fn run_impl(ctx: &Context, opts: CommandGlobalOpts, cmd: ShowCommand) -> miette::Result<()> {
-    let node = InMemoryNode::start(ctx, &opts.state).await?;
-    let controller = node.create_controller().await?;
+pub struct ProjectShowTui {
+    ctx: Context,
+    opts: CommandGlobalOpts,
+    project_name: Option<String>,
+}
+impl ProjectShowTui {
+    pub async fn run(
+        ctx: Context,
+        opts: CommandGlobalOpts,
+        project_name: Option<String>,
+    ) -> miette::Result<()> {
+        let tui = Self {
+            ctx,
+            opts,
+            project_name,
+        };
+        tui.show().await
+    }
+}
+
+#[ockam_core::async_trait]
+impl ShowCommandTui for ProjectShowTui {
+    const ITEM_NAME: &'static str = "projects";
```

Comment:
This should be in singular. This part is a little hacky right now and I plan to add both variants to the trait.